src/main/java/io/github/thebusybiscuit/slimefun4/api/events/AutoEnchantProcessEvent.java

```diff
@@ -0,0 +1,86 @@
+package io.github.thebusybiscuit.slimefun4.api.events;
+
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.machines.enchanting.AutoEnchanter;
+import me.mrCookieSlime.Slimefun.api.inventory.BlockMenu;
+
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+
+/**
+ * An {@link Event} that is called whenever an {@link AutoEnchanter} is
+ * enchanting an {@link ItemStack}.
+ *
+ * @author StarWishsama
+ */
+public class AutoEnchantProcessEvent extends Event implements Cancellable {
+    private static final HandlerList handlers = new HandlerList();
+
+    private final ItemStack item;
+    private final ItemStack enchantBook;
+    private final BlockMenu menu;
+
+    private boolean cancelled;
+
+    public AutoEnchantProcessEvent(@Nonnull ItemStack item, @Nonnull ItemStack enchantBook, @Nonnull BlockMenu menu) {
+        super(true);
```

Comment:
Since you're allowing this to run on another thread, the event should be named `AsyncAutoEnchantProcessEvent` to keep with Bukkit standards